core/tauri-utils/src/config.rs

```diff
@@ -533,6 +533,48 @@ fn default_allow_downgrades() -> bool {
   true
 }
 
+/// macOS-only. Corresponds to CFBundleTypeRole
+#[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize)]
+#[cfg_attr(feature = "schema", derive(JsonSchema))]
+pub enum BundleTypeRole {
+  /// CFBundleTypeRole.Editor. Files can be read and edited.
+  Editor,
+  /// CFBundleTypeRole.Viewer. Files can be read.
+  Viewer,
+  /// CFBundleTypeRole.Shell
+  Shell,
+  /// CFBundleTypeRole.QLGenerator
+  QLGenerator,
+  /// CFBundleTypeRole.None
+  None,
+}
+
+impl Display for BundleTypeRole {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    match self {
+      Self::Editor => write!(f, "Editor"),
+      Self::Viewer => write!(f, "Viewer"),
+      Self::Shell => write!(f, "Shell"),
+      Self::QLGenerator => write!(f, "QLGenerator"),
+      Self::None => write!(f, "None"),
+    }
+  }
+}
+
+/// File association
+#[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize)]
+#[cfg_attr(feature = "schema", derive(JsonSchema))]
+pub struct FileAssociation {
+  /// File extensions to associate with this app. e.g. 'png'
+  pub ext: Vec<String>,
+  /// The name. Default to ext[0]
+  pub name: Option<String>,
+  /// macOS-only. The app’s role with respect to the type.
+  pub role: Option<BundleTypeRole>,
+  /// Linux-only. The mime-type. e.g. 'image/png'
+  pub mime_type: Option<Vec<String>>,
```

Comment:
I think it should be avoided to have elements in the vectors `ext` and `mime_type` connected, I'd prefer `Vec<{ext, name}>` personally